src/main/java/org/kpmp/geneExpressionSummary/GeneExpressionSummaryService.java

```diff
@@ -58,16 +64,33 @@ public List<? extends GeneExpressionSummary> getByDataTypeEnrollmentCategoryAndG
 					.findByEnrollmentAndGeneAllClusters(geneSymbol, enrollmentCategoryEnum.getParticipantEnrollmentCategory()).stream()
 					.distinct().collect(Collectors.toList());
 		case SINGLE_NUCLEUS:
-			return snrnaGeneExpressionRepository
-					.findByEnrollmentAndGeneAllClusters(geneSymbol, enrollmentCategoryEnum.getParticipantEnrollmentCategory()).stream()
-					.distinct().collect(Collectors.toList());
+            if(newData != null && newData){
+                return snrnaGeneExpressionRepositoryNewData
+                        .findByEnrollmentAndGeneAllClusters(geneSymbol, enrollmentCategoryEnum.getParticipantEnrollmentCategory()).stream()
+                        .distinct().collect(Collectors.toList());
+            }
+            else{
+                return snrnaGeneExpressionRepository
+                .findByEnrollmentAndGeneAllClusters(geneSymbol, enrollmentCategoryEnum.getParticipantEnrollmentCategory()).stream()
+                .distinct().collect(Collectors.toList());
+            }
 		case UNKNOWN:
-			List<GeneExpressionSummary> allResults = new ArrayList<>();
-			allResults.addAll(scrnaGeneExpressionRepository.findByEnrollmentAndGeneAllClusters(geneSymbol,
-					enrollmentCategoryEnum.getParticipantEnrollmentCategory()));
-			allResults.addAll(snrnaGeneExpressionRepository.findByEnrollmentAndGeneAllClusters(geneSymbol,
-					enrollmentCategoryEnum.getParticipantEnrollmentCategory()));
-			return allResults;
+            if (newData != null && newData) {

```

Comment:
@Dert1129 : FYI, if you treated this as a primative boolean, you wouldn't have to do a null check.